### PR TITLE
[pytorch] use correct warning type for tracer warnings

### PR DESCRIPTION
**Pull request by ppwwyyxx**
Summary:
We have code to ignore this category of warnings and found this one is incorrect.

Use `stacklevel=2`, otherwise the warning is always filtered by TracerWarning.ignore_lib_warnings()

Test Plan: sandcastle

Differential Revision: D26867290

### CODE REVIEW

**Review by wanchaol — APPROVED** [review]
looks good, thanks!

File: torch/tensor.py [hunk 1]
```diff
@@ -593,7 +593,7 @@ def __iter__(self):
             warnings.warn('Iterating over a tensor might cause the trace to be incorrect. '
                           'Passing a tensor of different shape won\'t change the number of '
                           'iterations executed (and might lead to errors or silently give '
-                          'incorrect results).', category=RuntimeWarning)
+                          'incorrect results).', category=torch.jit.TracerWarning, stacklevel=2)
```
Comment (albanD) [inline torch/tensor.py:596]: Why should this one be a TracerWarning?  That doesn't look related to jit tracing at all.

Comment (ppwwyyxx) [reply torch/tensor.py:596]: The warning only issues during tracing, and is about how the trace will not generalize to a different shape.
